File: app/src/main/java/com/billpayments/features/billers/data/BillerCatalog.kt
package com.billpayments.features.billers.data

data class BillerCatalog(
    val categories: List<BillerCategory>,
    val billers: List<Biller>,
) {
    fun billersFor(categoryId: String): List<Biller> = billers.filter { it.categoryId == categoryId }

    fun billerById(id: String): Biller? = billers.firstOrNull { it.id == id }

    fun categoryById(id: String): BillerCategory? = categories.firstOrNull { it.id == id }
}

File: BillPayments/Features/Billers/Data/BillerCatalog.swift
struct BillerCatalog: Equatable {
    let categories: [BillerCategory]
    let billers: [Biller]

    func billersFor(categoryId: String) -> [Biller] {
        billers.filter { $0.categoryId == categoryId }
    }

    func billerById(_ id: String) -> Biller? {
        billers.first { $0.id == id }
    }

    func categoryById(_ id: String) -> BillerCategory? {
        categories.first { $0.id == id }
    }
}
